### PR TITLE
Highlight: client can provide an linking policy

### PR DESCRIPTION
**Pull request by privat**
Links to entities can be defined by specializing `HighlightVisitor::hrefto`

Nitlight uses it to correctly provide better links (and avoid generating dead links)

### CODE REVIEW

**Comment by privat** [comment]
ping @Morriar

**Comment by privat** [comment]
demo is up http://info.uqam.ca/~privat/tmp/nitlight/

File: src/highlight.nit [hunk 1]
```diff
@@ -395,51 +401,63 @@ end
...
+	#
+	# For some complex entities, like generic types, multiple `<a>` and `<span>` elements can be generated.
+	# E.g. `Array[Int]` might become `<a>Array</a>[<a>Int</a>]` with the correct `href` attributes
+	# provided  by `v.hrefto`.
+	fun linkto(v: HighlightVisitor): HTMLTag do return linkto_text(v, to_s)
+
+	# Link to the `self` with a specific text
+	#
+	# The whole text is linked with a single `<a>` element.
+
```
Comment (jcbrinfo) [inline src/highlight.nit]: Broken comment.

**Comment by jcbrinfo** [comment]
Less coupling with the same amount of code.
:shipit: (once the documentation of `linkto_text` is fixed)

**Comment by Morriar** [comment]
+1

**Comment by privat** [comment]
reroll fix broken comment

**Pull request merged**